src/main/java/bio/terra/cli/service/GoogleOauth.java

```diff
@@ -252,19 +401,39 @@ public static ServiceAccountCredentials getServiceAccountCredential(
   /**
    * Refresh the credential if expired and then return its access token.
    *
-   * @param credential credentials object
+   * @param credential Terra credentials object
    * @return access token
    */
-  public static AccessToken getAccessToken(GoogleCredentials credential) {
+  public static AccessToken getAccessToken(TerraCredentials credential) {
     try {
-      credential.refreshIfExpired();
+      credential.getGoogleCredentials().refreshIfExpired();
     } catch (IOException ioEx) {
       logger.warn("Error refreshing access token", ioEx);
       // don't throw an exception here because the token may not be expired, in which case it's fine
       // to use it without refreshing first. if the token is expired, then we'll get a permission
       // error when we try to re-use it anyway, and this log statement may help with debugging.
     }
-    return credential.getAccessToken();
+    return credential.getGoogleCredentials().getAccessToken();
+  }
+
+  /**
+   * @param credentials Terra credentials object
+   * @return id token
+   */
+  public static IdToken getIdToken(TerraCredentials credentials) {
+    IdToken idToken = credentials.getIdToken();
+    long expiration = idToken.getExpirationTime().getTime();
+    long now = System.currentTimeMillis();
+    if (now > expiration) {
+      // If we get here, there is no action we can take since we can't refresh an ID token, and no
```

Comment:
Is there anything about an expired token that's useful to the caller? If not, why do we return it?